framework/src/problems/FEProblemBase.C

```diff
@@ -3762,7 +3778,11 @@ FEProblemBase::execMultiAppTransfers(ExecFlagType type, MultiAppTransfer::DIRECT
     _console << COLOR_CYAN << "\nStarting Transfers on " << Moose::stringify(type)
              << string_direction << "MultiApps" << COLOR_DEFAULT << std::endl;
     for (const auto & transfer : transfers)
-      transfer->execute();
+    {
+      const auto multi_app_transfer = std::dynamic_pointer_cast<MultiAppTransfer>(transfer);
```

Comment:
I would argue that direction and associated methods/attributes should be moved up to the base class here. The idea of a direction makes sense at the transfer level. It would also mean that you wouldn't have to do all this dynamic casting when you need to execute. I'm pretty sure there aren't any pure "Transfers" in MOOSE. @friedmud put that in in case we needed it but in practice all Transfers are MultiappTransfers. Even if they aren't, that doesn't change my first statement.